node/router/src/helpers/resolver.rs

```diff
@@ -13,15 +13,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use parking_lot::RwLock;
 use std::{collections::HashMap, net::SocketAddr};
 
 #[derive(Debug)]
 pub struct Resolver {
     /// The map of the listener address to (ambiguous) peer address.
-    from_listener: RwLock<HashMap<SocketAddr, SocketAddr>>,
+    from_listener: HashMap<SocketAddr, SocketAddr>,
```

Comment:
Not a change request, but one thing I wanted to do eventually is introduce a type `struct PeerId(SocketAddr)`
This would make it clearer that we use the listener address as a unique identifier and prevent accidental/implicit conversions between addresses and IDs.